pyinfra/api/connectors/ssh.py

```diff
@@ -420,8 +420,25 @@ def put_file(
         temp_file = state.get_temp_filename(remote_filename)
         _put_file(host, filename_or_io, temp_file)
 
+        # Make sure our sudo/su user can access the file
+        if su_user:
+            command = StringCommand('setfacl -m u:{0}:r'.format(su_user), temp_file)
+        elif sudo_user:
+            command = StringCommand('setfacl -m u:{0}:r'.format(sudo_user), temp_file)
+        status, _, stderr = run_shell_command(
+            state, host, command,
+            sudo=False,
+            print_output=print_output,
+            print_input=print_input,
+            **command_kwargs
+        )
+
+        if status is False:
+            logger.error('Error on handover to sudo/su user: {0}'.format('\n'.join(stderr)))
+            return False
+
         # Execute run_shell_command w/sudo and/or su_user
-        command = StringCommand('mv', temp_file, QuoteString(remote_filename))
+        command = StringCommand('cp', temp_file, QuoteString(remote_filename))
```

Comment:
```suggestion
        command = StringCommand('cp', temp_file, QuoteString(remote_filename), '&&', 'rm', '-f', temp_file)
```

I think it makes sense to remove the temporary file here once the copy is complete.

Reply:
I think for this one we will need to do the removal outside the sudo session, otherwise we'll end up with the same problem as `mv` where we can't delete it due to the sticky bit. I'll give something a try tomorrow and make sure it works!

Reply:
I've added an `rm` outside the `sudo` session to remove the file now.